xbmc/platform/darwin/osx/network/NetworkMacOS.mm

```diff
@@ -0,0 +1,304 @@
+                                    cp[2], cp[3], cp[4], cp[5]);
+          ret = true;
+          break;
+        }
+      }
+      free(buf);
+    }
+  }
+  return ret;
+}
+
+std::unique_ptr<CNetworkBase> CNetworkBase::GetNetwork()
+{
+  return std::make_unique<CNetworkMacOS>();
+}
+
+CNetworkMacOS::CNetworkMacOS() : CNetworkPosix()
+{
+  queryInterfaceList();
+}
+
+void CNetworkMacOS::GetMacAddress(const std::string& interfaceName, char rawMac[6])
+{
+  memset(rawMac, 0, 6);
+
+#if !defined(IFT_ETHER)
+#define IFT_ETHER 0x6 /* Ethernet CSMACD */
+#endif
+  const struct sockaddr_dl* dlAddr = NULL;
+  const uint8_t* base = NULL;
+  // Query the list of interfaces.
+  struct ifaddrs* list;
+  struct ifaddrs* interface;
+
+  if (getifaddrs(&list) < 0)
+  {
+    return;
+  }
+
+  for (interface = list; interface != NULL; interface = interface->ifa_next)
+  {
+    if (interfaceName == interface->ifa_name)
+    {
+      if ((interface->ifa_addr->sa_family == AF_LINK) &&
+          (((const struct sockaddr_dl*)interface->ifa_addr)->sdl_type == IFT_ETHER))
+      {
+        dlAddr = (const struct sockaddr_dl*)interface->ifa_addr;
+        base = (const uint8_t*)&dlAddr->sdl_data[dlAddr->sdl_nlen];
+
+        if (dlAddr->sdl_alen > 5)
+        {
+          memcpy(rawMac, base, 6);
+        }
+      }
+      break;
+    }
+  }
+
+  freeifaddrs(list);
+}
+
+void CNetworkMacOS::queryInterfaceList()
+{
+  char macAddrRaw[6];
+  m_interfaces.clear();
+
+  // Query the list of interfaces.
+  struct ifaddrs* list;
+  if (getifaddrs(&list) < 0)
+    return;
+
+  struct ifaddrs* cur;
+  for (cur = list; cur != NULL; cur = cur->ifa_next)
+  {
+    if (cur->ifa_addr->sa_family != AF_INET)
+      continue;
+
+    GetMacAddress(cur->ifa_name, macAddrRaw);
+
+    // only add interfaces with non-zero mac addresses
+    if (macAddrRaw[0] || macAddrRaw[1] || macAddrRaw[2] || macAddrRaw[3] || macAddrRaw[4] ||
+        macAddrRaw[5])
+      // Add the interface.
+      m_interfaces.push_back(new CNetworkInterfaceMacOS(this, cur->ifa_name, macAddrRaw));
+  }
+
+  freeifaddrs(list);
+}
+
+std::vector<std::string> CNetworkMacOS::GetNameServers()
+{
+  __block std::vector<std::string> result;
+  @autoreleasepool
+  {
+    // Create an SCPreferencesRef object for the system preferences
+    SCPreferencesRef preferences = SCPreferencesCreate(nullptr, CFSTR("GetDNSInfo"), nullptr);
+    if (!preferences)
+    {
+      CLog::Log(LOGWARNING, "Unable to determine nameservers");
+      return result;
+    }
+
+    // Get the current DNS configuration from system preferences
+    CFDictionaryRef currentDNSConfig =
+        static_cast<CFDictionaryRef>(SCPreferencesGetValue(preferences, kSCPrefNetworkServices));
+    if (currentDNSConfig)
+    {
+      NSDictionary* networkServices = (__bridge NSDictionary*)currentDNSConfig;
+      [networkServices enumerateKeysAndObjectsUsingBlock:^(id key, id obj, BOOL* stop) {
+        NSDictionary* service = networkServices[key];
```

Comment:
`obj` is exactly your `service` ;) you can also use proper types in the block declaration instead of `id`